.github/workflows/ci.yml

```
@@ -3,89 +3,110 @@ name: CI
+    # Possibly resolveable with https://github.com/actions/runner/issues/320
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Run ROS Industrial CI
-        uses: ros-industrial/industrial_ci@master
-        env: ${{matrix.env}}
 
-  catkin_lint:
-    strategy:
-      matrix:
-        env:
-          - { ROS_DISTRO: melodic, ROS_REPO: main, CATKIN_LINT: pedantic, CATKIN_LINT_ARGS: "--config $TARGET_REPO_PATH/.catkin-lint", TARGET_WORKSPACE: "$TARGET_REPO_PATH metapackage_dependencies.rosinstall", UPSTREAM_WORKSPACE: "upstream_dependencies.rosinstall" }
-
-    runs-on: ubuntu-18.04
-    steps:
-      - name: Checkout repository
+      - name: Checkout Repository
         uses: actions/checkout@v2
 
-      - name: Run ROS Industrial CI
-        uses: ros-industrial/industrial_ci@master
-        env: ${{matrix.env}}
+      - name: Install Docker/QEMU dependencies
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: arm64
 
-  arm64_build:
-    strategy:
-      matrix:
-        env:
-          - {DOCKER_IMAGE: "arm64v8/ros:melodic-perception", ADDITIONAL_DEBS: "build-essential python-catkin-tools python-pip", BEFORE_INIT: "echo -e This test should run on aarch64. It is running on: $(uname -p) && [[ $(uname -p) == aarch64 ]]", CMAKE_ARGS: "-Wall -Wextra -Wpedantic -Werror", TARGET_WORKSPACE: "$TARGET_REPO_PATH metapackage_dependencies.rosinstall", UPSTREAM_WORKSPACE: "upstream_dependencies.rosinstall"}
-
-    runs-on: ubuntu-18.04
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
+      - name: Setup Docker to Run ARM64 Containers in QEMU
+        run: |
+          docker run --privileged --rm multiarch/qemu-user-static --reset --credential yes --persistent yes
 
-      - name: Install Qemu and Start Qemu Docker
-        shell: bash
+      - name: Start ARM64 ROS2 Container
         run: |
-          sudo apt update
-          sudo apt install qemu-user-static
-          docker run --rm --privileged multiarch/qemu-user-static --reset --credential yes --persistent yes
+          docker run --name arm64-ros --mount type=bind,source=$(pwd),target=/rosws/src/$(basename $(pwd)) --workdir /rosws --detach -i --rm arm64v8/ros:foxy
 
-      - name: Run ROS Industrial CI
-        uses: ros-industrial/industrial_ci@master
-        env: ${{matrix.env}}
+      - name: Install ROS Package Dependencies
+        run: |
+          docker exec arm64-ros sudo apt update
+          docker exec arm64-ros rosdep install -r --from-paths . --ignore-src --rosdistro foxy -y
+        # todo foxy
+
+      - name: Build ROS2 Packages and Run Tests
+        run: >
+          docker exec arm64-ros bash -c "source /opt/ros/foxy/setup.bash;
+          colcon build --event-handlers console_cohesion+;
+          colcon test --event-handlers console_cohesion+"
```

```suggestion
        run: >
          docker exec arm64-ros bash -c
          "source /opt/ros/foxy/setup.bash;
          colcon build --event-handlers console_cohesion+;
          colcon test --event-handlers console_cohesion+"
```

fyi using `>` instead of `|` folds newlines into spaces